compute_modules/function_registry/function_schema_parser.py

```diff
@@ -198,8 +198,11 @@ def _extract_data_type(type_hint: typing.Any) -> typing.Tuple[DataTypeDict, Pyth
                 "elementsType": element_type,
             },
         }, PythonClassNode(constructor=list, children={"list": element_class_node})
-    if typing.get_origin(type_hint) is dict:
-        key_type, value_type = typing.get_args(type_hint)
+    if type_hint is dict or typing.get_origin(type_hint) is dict:
+        dict_type_hints = typing.get_args(type_hint)
+        if not len(dict_type_hints):
+            raise ValueError("dict type hints must have type parameters provided")
```

Comment:
From a. user perspective I am not sure about the message here. I think we are going to get a lot of questions on what this means for a user. So maybe add an example such as put `Dict[str,str]  instead of dict `
<img width="900" alt="Screenshot 2024-10-16 at 1 22 47 PM" src="https://github.com/user-attachments/assets/69fada58-af60-4fd8-8c8c-184c287a6383">